judge/views/problem.py

```diff
@@ -684,6 +724,43 @@ def form_valid(self, form):
         return HttpResponseRedirect(reverse('admin:judge_problem_change', args=(problem.id,)))
 
 
+class ProblemSuggest(TitleMixin, CreateView):
+    template_name = 'problem/suggest.html'
+    model = Problem
+    form_class = ProblemSuggestForm
+
+    def get_title(self):
+        return _('Suggesting new problem')
+
+    def get_content_title(self):
+        return _('Suggesting new problem')
+
+    def post(self, request, *args, **kwargs):
+        form = ProblemSuggestForm(request.POST or None)
+        if form.is_valid():
+            problem = form.save()
+            problem.suggester.add(request.user.profile)
+            problem.save()
+            return self.form_valid(form)
+        else:
+            return self.form_invalid(form)
```

Comment:
- Please add `self.object = None` at the beginning of the `post` method. This will prevent error when `form_invalid` get called.
- `problem.suggester.add(request.user.profile)` -> `problem.suggester = request.user.profile`
- Don't call `self.form_valid(form)` after call `form.save()` because the default `self.form_valid(form)` will automatically call `form.save()`.  You can simply `return return HttpResponseRedirect(self.get_success_url())`